Kernel/Scheduler.cpp

```diff
@@ -48,12 +50,26 @@ Thread* g_last_fpu_thread;
 Thread* g_finalizer;
 static Process* s_colonel_process;
 u64 g_uptime;
-static u64 s_beep_timeout;
+static u64 s_next_timer_due;
 
 struct TaskRedirectionData {
     u16 selector;
     TSS32 tss;
 };
+
+struct Timer {
```

Comment:
Timer objects should have some kind of identity so we can find them. It's okay for their address to be the identity, but then we should make sure their address is unique. The easiest way to do that is always allocating them on the heap.

We could also give them a timer ID, e.g an int or whatever seems best. :)